operate/schema/src/main/java/io/camunda/operate/util/OpensearchUtil.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.operate.util;
+
+import io.camunda.operate.exceptions.ArchiverException;
+import io.camunda.operate.exceptions.OperateRuntimeException;
+import java.io.IOException;
+import java.util.concurrent.CompletableFuture;
+import org.opensearch.client.opensearch.OpenSearchAsyncClient;
+import org.opensearch.client.opensearch.core.DeleteByQueryRequest;
+import org.opensearch.client.opensearch.core.DeleteByQueryResponse;
+import org.opensearch.client.opensearch.core.ReindexRequest;
+import org.opensearch.client.opensearch.core.ReindexResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class OpensearchUtil {
+  private static final Logger LOGGER = LoggerFactory.getLogger(OpensearchUtil.class);
+
+  public static <T> Either<Throwable, Long> handleResponse(
+      final T response, final Throwable error, final String sourceIndexName) {
```

Comment:
what about making this method more generic and avoid handling response types
```suggestion
public static Either<Throwable, Long> handleResponse(
    final Long total,
    final List<?> failures,
    final String operation,
    final Throwable error,
    final String sourceIndexName) {
```